pkg/api/v1/types.go

```diff
@@ -437,8 +440,21 @@ const (
 
 	// MountOptionAnnotation defines mount option annotation used in PVs
 	MountOptionAnnotation = "volume.beta.kubernetes.io/mount-options"
+
+	// StorageTopologyConstraint defines topology constraints of a PV
+	// Value is type []TopologyConstraint
+	AlphaStorageTopologyConstraintsAnnotation = "volume.alpha.kubernetes.io/topology-constraints"
 )
 
+// TopologyConstraint contains label information that the resource is constrained to
+// A label key can have multiple values
+type TopologyConstraint struct {
+	// Key of the label
```

Comment:
I'm not suggesting adding code for validation. But, the documentation should suggest that there are some special topology keys that are supported by upstream by default.
`Hostname` is not portable right? Don't we need `NodeName`?